x-pack/legacy/plugins/apm/server/lib/transactions/get_transaction/index.ts

```diff
@@ -45,3 +45,44 @@ export async function getTransaction(
   const resp = await client.search<Transaction>(params);
   return resp.hits.hits[0]?._source;
 }
+
+export async function getTransactionByTraceId(traceId: string, setup: Setup) {
+  const { client, indices } = setup;
+  const params = {
+    index: indices['apm_oss.transactionIndices'],
+    body: {
+      _source: ['timestamp', 'transaction', 'service.name'],
+      size: 1,
+      query: {
+        bool: {
+          should: [
+            {
+              constant_score: {
+                filter: {
+                  bool: {
+                    must_not: {
+                      exists: {
+                        field: 'parent.id'
+                      }
+                    }
+                  }
+                }
+              }
+            }
+          ],
+          filter: [
+            {
+              term: { 'trace.id': traceId }
+            },
+            {
+              term: { 'processor.event': 'transaction' }
+            }
+          ]
+        }
+      }
+    }
+  };
+
+  const resp = await client.search<Transaction>(params);
+  return resp.hits;
```

Comment:
What about just returning the transaction?
```suggestion
  return resp.hits.hits[0];
```

Reply:
I'm returning the hits because I'm checking the total hits, in order to redirect to the Trace page if any transaction is found. https://github.com/elastic/kibana/pull/51450/files#diff-b97f6df9a1e502bc5afd074ddbcf40cbR89

Reply:
I will return the transaction and use the `status` to check it instead.